Automattic/jetpack
# Content Options: Respect Block Settings for Featured Images

Fixes #29286
Fixes Automattic/wp-calypso#85672
Fixes Automattic/wp-calypso#85369
Fixes Automattic/wp-calypso#59280
Fixes Automattic/wp-calypso#73069
Fixes Automattic/wp-calypso#86094
Fixes Automattic/themes#7580
Fixes Automattic/themes#3255
Fixes Automattic/themes#3462
Fixes Automattic/themes#6034
Fixes Automattic/themes#3609
Fixes Automattic/themes#1943
Fixes Automattic/newspack-blocks#399
Fixes Automattic/newspack-blocks#851 

(that's a lot!)

## Proposed changes:

* Ensures that the Content Options' Featured Images selection doesn't override the user's selection for the Latest Posts, Homepage Articles, Post Carousel and Query Loop blocks.

**Update:** rather than using CSS as I first did, I've used the approach recommended by @anomiex here: https://github.com/Automattic/jetpack/issues/29286#issuecomment-1876043341. 

### Other information:


## Jetpack product discussion


N/A.

## Does this pull request change what data or activity we track or use?


No.

## Testing instructions:

* Go to the WordPress.com theme directory and install any theme which uses Content Options (any child theme of Varia, such as Stow, will work - but you can also use others like Canard) - https://wordpress.com/theme/stow
* Under the "Content Options" settings in the Customizer, hide Featured Images for posts or pages
* Create a post or a page with a Latest Posts block, the a8c Blog Posts block or the Query Loop block, and ensure that thumbnails are set to be displayed
* Set a Featured Image for that post or page
* Confirm that the Featured Image is hidden, but that the thumbnails within the blocks aren't
